### ISSUE TITLE
TOC Generator, docs: update TOC

### ISSUE THREAD

**Issue by akirashirosawa** [CleverRaven/Cataclysm-DDA#47262]
# Describe the bug
Yesterday, while I was making out the pull request, a strange commit appeared in the PR-branch, which I did not make.
I couldn't figure out where it came from and removed the branch and re-created it.
Today a similar commit appeared on the master branch of my fork.
Here it is: https://github.com/akirashirosawa/Cataclysm-DDA/commit/05bf4f0295e887f96f648c3db961cf8c48c291c8
Quick google says this is related with `GitHub Action`. I have not configured any `GitHub Action`.
What should I do about it? How to disable so as not to break consistency with the main repository?
# Steps To Reproduce
Fork and wait a while? Maybe do `git push` from local terminal (to push after `git pull --ff-only upstream master`).
# Expected behavior
No commits on my fork not made by me.
# Screenshots
# Versions and configuration
git version 2.30.0
# Additional context
I remembered that there was such a commit yesterday in the master branch too, I just made a force push. But I want to solve it normally.

**Comment by BrettDong** [comment]
I think it's #46820 activated.

**Comment by akirashirosawa** [comment]
Thanks. Now the reason is clear.
I read the github [documentation](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context) a bit. Apparently, we need to add a condition like:
```yml
on:
  push:
    branches:
    - master
    paths:
    - 'doc/JSON_INFO.md'
name: TOC Generator
jobs:
  generateTOC:
    if: github.repository_owner == "CleverRaven" <---- HERE ---
    name: TOC Generator
    runs-on: ubuntu-latest
    steps:
      - uses: technote-space/toc-generator@v2.6.1
        with:
          TARGET_PATHS: "doc/JSON_INFO.md"
          COMMIT_NAME: "TOC Generator"
          COMMIT_EMAIL: "noreply@example.invalid"
```
I'm not sure if it will work, I don't know how to test it adequately. But based on the documentation, it should.
Notes: 
1. The condition is often used: `github.event.pull_request.head.user.id == github.event.pull_request.base.user.id`. But this is not suitable for this repository, since the user creating the PR is not equal to the user who owns the branch.
2. Everywhere I've seen used: `technote-space/toc-generator@v3`, not `@v2.6.1`.

**Comment by akirashirosawa** [comment]
@hexagonrecursion please, look this issue.

**Comment by hexagonrecursion** [comment]
# I'm observing two discrepancies
## First
* 22 minutes ago a PR was merged that touched doc/JSON_INFO.md https://github.com/CleverRaven/Cataclysm-DDA/commit/2fb49d34cdf6894883288bfccd7effc783604d77
* https://github.com/CleverRaven/Cataclysm-DDA/commits/master/doc/JSON_INFO.md currently claims that the latest commit to doc/JSON_INFO.md was made 11 hours ago
## Second
* https://github.com/CleverRaven/Cataclysm-DDA/runs/1847881575?check_suite_focus=true#step:2:574 claims that the TOC Generator workflow
* I don't see a commit it created on master https://github.com/CleverRaven/Cataclysm-DDA/commits/master
* Did someone force-push?

**Comment by hexagonrecursion** [comment]
# Regarding the second discrepancy
It looks like technote-space/toc-generator@v2.6.1 has a bug: it reports success even if it fails due to branch protection
https://github.com/hexagonrecursion/toc/runs/1847959476?check_suite_focus=true#step:2:428

**Comment by BrettDong** [comment]
I think your workflow is unable to commit directly on master branch because permission issues.

**Comment by hexagonrecursion** [comment]
Reported upstream bug https://github.com/technote-space/toc-generator/issues/311

**Comment by Zireael07** [comment]
Have you tried updating the toc-generator action to v3 first?

**Comment by kevingranade** [comment]
Should be fixed by #47302
